example/binary/consumer/tests/Service/HttpClientServiceTest.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace BinaryConsumer\Tests\Service;
+
+use BinaryConsumer\Service\HttpClientService;
+use PhpPact\Consumer\InteractionBuilder;
+use PhpPact\Consumer\Model\Body\Binary;
+use PhpPact\Consumer\Model\ConsumerRequest;
+use PhpPact\Consumer\Model\ProviderResponse;
+use PhpPact\Standalone\MockService\MockServerConfig;
+use PHPUnit\Framework\TestCase;
+
+class HttpClientServiceTest extends TestCase
+{
+    public function testGetImageContent()
+    {
+        $imageContent = file_get_contents(__DIR__ . '/../_resource/image.jpg');
+
+        $request = new ConsumerRequest();
+        $request
+            ->setMethod('GET')
+            ->setPath('/image.jpg')
+            ->addHeader('Accept', 'image/jpeg');
+
+        $response = new ProviderResponse();
+        $response
+            ->setStatus(200)
+            ->addHeader('Content-Type', 'image/jpeg')
+            ->setBody(new Binary($imageContent, in_array(php_uname('m'), ['AMD64', 'arm64']) ? 'application/octet-stream' : 'image/jpeg'));
```

Comment:
Related to https://github.com/pact-foundation/pact-reference/issues/305 I presume

Reply:
I think https://github.com/pact-foundation/pact-reference/issues/171 is more precise

Reply:
ahh yes, spot on 🔍